samples/bot-conversation/csharp/Bots/TeamsConversationBot.cs

```diff
@@ -72,6 +80,63 @@ protected override async Task OnTeamsMembersAddedAsync(IList<TeamsChannelAccount
             }
         }
 
+        /// <summary>
+        /// Checks the count of members who have read the message sent by MessageAllMembers command
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private async Task CheckReadUserCount(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            if (teamMemberDetails.Count != 0)
+            {
+                int count = 0;
+                List<string> users = new List<string>();
+                foreach (var memberDetails in teamMemberReadDetails)
+                {
+                    if (teamMemberDetails.ContainsKey(memberDetails.Key))
+                    {
+                        count++;
+                        teamMemberDetails.TryGetValue(memberDetails.Key, out TeamsChannelAccount value);
+                        users.Add(value.Name);
+                    }
+                }
+
+                var userList = string.Join(", ", users);
+                await turnContext.SendActivityAsync(MessageFactory.Text($"Number of members read the message : {count} \n\n Members : {userList}"), cancellationToken);
+            }
+            else
+            {
+                await turnContext.SendActivityAsync(MessageFactory.Text("Read count is zero. Please make sure to send message to all members firstly to check the count of members who have read your message."), cancellationToken);
+            }
+
+        }
+
+        /// <summary>
+        /// Resets the check count of members who have read the message sent by MessageAllMembers command
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        private async Task ResetReadUserCount(ITurnContext<IMessageActivity> turnContext, CancellationToken cancellationToken)
+        {
+            teamMemberDetails = new ConcurrentDictionary<string, TeamsChannelAccount>();
+            teamMemberReadDetails = new ConcurrentDictionary<string, ChannelAccount>();
+        }
+
+        /// <summary>
+        /// Invoked when user read the message sent by bot in personal scope
+        /// </summary>
+        /// <param name="turnContext"></param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        protected override async Task OnTeamsReadReceiptAsync(ReadReceiptInfo readReceiptInfo, ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
```

Comment:
Can we show the example of using IsMessageRead? 

On the bot side, the bot developer will need to store one id of a bot's message (the message that a developer cares). ex: A company communicator bot sends an announcement to 5 people in the tenant. It wants to see whether everyone read that message. After the bot sent the announcmenet message, save that message id .
And whenver it receives read receipt event, compare the saved message id and ReadReceiptInfo.lastReadMessageId

Reply:
Implemented this
After the bot sent the announcmenet message, save that message id .
And whenver it receives read receipt event, compare the saved message id and ReadReceiptInfo.lastReadMessageId

Reply:
THank you!

Reply:
@Jegadeesh-MSFT - Please confirm about this functionality: **IsMessageRead** and implement it accordingly.